en/computer_vision/obstacle_avoidance.md

```diff
@@ -0,0 +1,156 @@
+# Obstacle Avoidance
+
+*Obstacle Avoidance* enables a vehicle to navigate around obstacles when following a preplanned path.
+
+The feature requires a companion computer that is running computer vision software. 
+This software provides a route for a given desired trajectory, mapping and navigating around obstacles to achieve the best path.
+
+Obstacle avoidance is intended for automatic modes, and is currently supported for multicopter vehicles in [Missions](#mission_mode) and [Offboard mode](#offboard_mode).
+
+This topic explains how the feature is set up and enabled in both modes.
+
+
+## Offboard Mode Avoidance {#offboard_mode}
+
+PX4 supports collision avoidance in [Offboard mode](flight_modes/offboard.md).
+
+The desired route comes from a [ROS](http://dev.px4.io/en/ros/) node running on a companion computer.
+This is passed into an obstacle avoidance module (another ROS node).
+The avoidance software sends the planned path to the flight stack as a stream of `SET_POSITION_TARGET_LOCAL_NED` messages.
+
+> **Note** The only required PX4-side setup is to put PX4 into *Offboard mode*..
+  While the `SET_POSITION_TARGET_LOCAL_NED` setpoints come from a ROS collision avoidance node, to PX4 could be from any MAVLink system.
+
+The tested hardware/software platform is [Intel Aero](../flight_controller/intel_aero.md) running either the *local_planner* or *global_planner*. 
```

Comment:
OK, so my assumption here is that the instructions in PX4/avoidance are good enough for arbitrary person X to find out what hardware they need for avoidance, and how to set up the software for use with this system. I have my doubts :-)

I think it makes sense to offload that to the avoidance docs, because none of that is PX4, and lots of things might use that software and setup.

Reply:
What specific configuration do we need to do to get trajectory messages emitted?

Reply:
Do you mean hardware setup?

Reply:
@mrivi No, I mean software. For example, to get local planner to emit the right message for collision avoidance we have to use the *rqt_reconfigure* tool and set the parameter `send_obstacles_fcu` to true. 

That said, IMO this feature will remain completely unusable until some work is done on the PX4/avoidance page to better explain how the hardware is set up.

Reply:
it's default

Reply:
Thanks. Let's make this clear in global_planner docs.